Reporting/M5 Document/Reflection.md

```diff
@@ -0,0 +1,29 @@
+
+- In future projects, we would prioritize more frequent cross-team check-ins and perhaps consider adopting a more integrated development approach from the outset to mitigate such challenges.
+
+## Do you feel that your initial requirements were sufficiently detailed for this project?  Which requirements did you miss or overlook?
+- While our initial requirements provided a solid foundation for the project, there were some areas where additional detail could have been beneficial. Specifically, we overlooked the specifics of password security measures, and multimedia messaging capabilities. 
+
+- In the future, we would ensure a more comprehensive review of security and messaging features during the requirement gathering phase to avoid oversights.
+
+## What did you miss in your initial planning for the project (beyond just the requirements)?
+- In our initial planning, we may have overlooked the potential impact of external dependencies and integration challenges. Specifically, we did not anticipate the complexities associated with integrating encryption protocols into our messaging system or the time required for thorough testing across different platforms and devices.
+
+- In the future, we would allocate more time and resources to address these aspects during the planning phase to minimize delays during development.
+
+## Would you (as a team) deal with testing differently in the future?
+
+- Looking back, we recognize that our testing approach could have been more comprehensive. While we conducted unit tests and integration tests throughout the development process, we underestimated the importance of user acceptance testing (UAT) and end-to-end testing.
+
+- In the future, we would prioritize UAT sessions with real users to gather feedback early and address any usability issues proactively.
+
+## If you were to estimate the efforts required for this project again, what would you consider?  (Really I am asking the team to reflect on the difference between what you thought it would take to complete the project vs what it actually took to deliver it). 
+
+- When estimating the efforts required for this project initially, we primarily considered the complexity of the features, the experience level of our team members, and the projected timeline. However, in hindsight, we underestimated the time required for debugging and troubleshooting, particularly when integrating third-party libraries and implementing encryption functionalities.
+
+- In the future, we would factor in a buffer for unforeseen challenges and prioritize incremental releases to facilitate early feedback and course correction.
```

Comment:
I totally agree with the buffer idea! That would be excellent, but with a 1-week sprint forced upon us, that was never an option. Good idea for the future, however!